.circleci/config.yml

```diff
@@ -0,0 +1,86 @@
+---
+# "Include" for unit tests definition.
+unit_tests: &unit_tests
+  steps:
+    - checkout
+    - run:
+        name: Install modules and dependencies.
+        command: npm install
+    - run:
+        name: Run unit tests.
+        command: npm test
+    - run:
+        name: Submit coverage data to codecov.
+        command: npm run codecov
+        when: always
+
+version: 2.0
+workflows:
+  version: 2
+  tests:
+    jobs:
+      - node4:
+          filters:
+            tags:
+              only: /^v\d+(\.\d+){2}(-.*)?$/
```

Comment:
I learned a trick from @kjin for defining this regexp only once. https://github.com/google/node-h2-auto-push/blob/9b09ae27c31a637ae32a23c05957944a5d530ef6/.circleci/config.yml#L2

Reply:
Why do we need to filter by tag for these tasks at all?

Reply:
This is so that these tests also run for those tags as well as changes in branches. That's needed because `node4`, `node6`, etc. are requirements of `publish_npm`, which is triggered only for such tags.

CircleCI build is performed for branches by default, but is not for tags by default.

Reply:
hmm. we have different setup in our other repos (see my comment below)

Reply:
Woah this is rad.  Thanks!